docs/docs/training-data-format.mdx

```diff
@@ -428,6 +428,42 @@ policies learn to predict the next action based on a *combination* of
 both the intent and entities (you can, however, change this behavior
 using the [`use_entities`](#entities) attribute).
 
+If you use `user:` key while writing stories, you do not have to deal with the specific
+intents of the messages that are extracted by the NLU pipeline. Instead, you can simply put
+the text of the user message.
+We call it end-to-end training because actual text is used for predictions.
+
+:::caution
+This feature is experimental.
+We introduce experimental features to get feedback from our community, so we encourage you to try it out!
+However, the functionality might be changed or removed in the future.
+If you have feedback (positive or negative) please share it with us on the [Rasa Forum](https://forum.rasa.com).
+
+:::
+
+These end-to-end user messages follow the format:
+
+```yaml-rasa
+stories:
+- story: user message structure
+  steps:
+    - user: the actual text of the user message  # Required
```

Comment:
what does `# required` mean? What is it supposed to tell the user?

Reply:
I don't know, it was there

Reply:
removed it